src/sources/splunk_hec/mod.rs

```diff
@@ -657,6 +740,72 @@ impl<'de, R: JsonRead<'de>> EventIterator<'de, R> {
 
         Ok(log.into())
     }
+
+    /// Build the log event for the vector namespace.
+    /// In this namespace the log event is created entirely from the event field.
+    /// No renaming of the `line` field is done.
+    fn build_log_vector(&mut self, json: &mut JsonValue) -> Result<LogEvent, Rejection> {
+        match json.get("event") {
+            Some(event) => {
+                let event: Value = event.into();
+                let mut log = LogEvent::from(event);
+
+                // The timestamp is extracted from the message for the Legacy namespace.
+                self.log_namespace.insert_vector_metadata(
+                    &mut log,
+                    lookup::path!(log_schema().timestamp_key()),
+                    lookup::path!("ingest_timestamp"),
```

Comment:
Are we setting this multiple times, there's a `insert_standard_vector_source_metadata` on ln993

Reply:
Yeah, this is for the event endpoint. Line 993 is only for the raw endpoint.